sts/signer.go

```diff
@@ -274,11 +279,18 @@ func (s *Signer) SignRequest(req *http.Request) error {
 		}
 
 		var buf bytes.Buffer
+		host := req.URL.Hostname()
+
+		// Check if the host IP is in IPv6 format. If yes, add the opening and closing square brackets.
+		if govalidator.IsIPv6(host) {
```

Comment:
Rather that pull in a dependency on the govalidator package, we can check using Go's stdlib:
```go
func isIPv6(s string) bool {
	ip := net.ParseIP(s)
	if ip == nil {
		return false
	}
	return len(ip) == net.IPv6len
}
```

Reply:
Thanks! I removed the govalidator dependency and added the `isIPv6` function.